src/ucp/core/ucp_request.inl

```diff
@@ -673,9 +612,8 @@ ucp_send_request_set_user_memh(ucp_request_t *req, ucp_md_map_t md_map,
     /* req->send.state.dt should not be used with protov2 */
     ucs_assert(!req->send.ep->worker->context->config.ext.proto_enable);
 
-    ucs_assert(!(req->flags & UCP_REQUEST_FLAG_USER_MEMH));
-    ucp_request_init_dt_reg_from_memh(req, md_map, param->memh,
-                                      &req->send.state.dt.dt.contig);
+    req->send.state.dt.dt.contig.memh = param->memh;
+    req->send.state.dt.dt.contig.memh->super.refcount++;
```

Comment:
1. why needed - ucp_memh_get should add refcount?
2. can we use rcache API for this?